### DISCUSSION TITLE
Stopping particle tracking at different times

### DISCUSSION THREAD

**Discussion by MiriamSterl** [Parcels-code/Parcels#1344] [Q&A] [answered]
I want to run a simulation where I release particles from the same location at different times, but also stop tracking them at different times, so that they are all advected for the same amount of time. For example, the first set of particles is released at `t = 0` and stops at `t = 100`; the next set is released at `t = 25` and stops at `t = 125`; etc. I know that I can use the `repeatdt` argument to release the particles at different times, but is there an easy way to make them stop moving at different times as well?

**Comment by VeckoTheGecko** [comment, marked as answer]
I see two ways that this could be done. With kernels, or using post-processing with xarray.

## Custom kernels
You can add an "age" to particles (e.g. in seconds as shown below) which tracks how old each particle is. If above a threshold it gets deleted.

Here's an code snippet illustrating what I mean.
```python
class AgingParticle(JITParticle):
    # Age in seconds
    age = Variable(
        'age', dtype=np.float32,
        initial=0.0,
    )

def aging_kernel(particle, fieldset, time):
    particle.age = particle.age + particle.dt

def delete_old_particle_kernel(particle, fieldset, time):
    if particle.age > 25.0:
        particle.delete()

fieldset = ...
lon = ...
lat = ...
pset = ParticleSet(fieldset, pclass=AgingParticle, lon=lon, lat=lat)
combined_kernel = pset.Kernel(aging_kernel) + AdvectionRK4 + pset.Kernel(delete_old_particle_kernel)

...
```

## Post-processing
You can add a variable tracking age (as above), but rather than deleting the particle, you can keep the particle in the simulation. Then once the simulation is done, you can open the data in xarray and select the observations where `age <= 25` and throw away everything else during analysis.

This takes up more space and computation, but adds flexibility to not have to run the simulation again should you want to change the cutoff.

Let me know how you go, or if you want me to elaborate😄

**Reply by erikvansebille** [reply]
This would be exactly my suggestion too!
